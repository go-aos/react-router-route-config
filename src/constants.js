export const configName = 'reactRouter'
export const propKey = 'routeMap'
