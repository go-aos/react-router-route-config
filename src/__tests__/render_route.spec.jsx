import { MemoryRouter } from 'react-router-dom'
import React from 'react'
import TestRenderer from 'react-test-renderer'

import { configName as defaultConfigName } from '../constants'
import renderRoute from '../render_route'

describe('renderRoute', () => {
  const customConfigName = 'customConfigName'
  const config = {
    exact: true,
    // eslint-disable-next-line
    render: () => <div>Hello World!!!</div>
  }
  const route = {
    config: {
      [customConfigName]: config,
      [defaultConfigName]: config
    },
    id: 'home',
    key: 'home',
    fullPath: '/home'
  }
  const createRoute = children => TestRenderer.create(
    <MemoryRouter initialEntries={['/home']}>
      {children}
    </MemoryRouter>
  )
  const testByConfigName = (configName = defaultConfigName) => {
    const routeElement = renderRoute(route, { configName })
    expect(routeElement.props).toMatchObject({
      exact: true,
      path: route.fullPath
    })
    const renderedRoute = createRoute(routeElement)
    const resJSON = renderedRoute.toJSON()
    expect(resJSON).toEqual(TestRenderer.create(config.render()).toJSON())
    expect(resJSON).toMatchSnapshot()
  }

  test('render component using default config name', () => {
    testByConfigName()
  })

  test('render component using custom config name', () => {
    testByConfigName(customConfigName)
  })
})
