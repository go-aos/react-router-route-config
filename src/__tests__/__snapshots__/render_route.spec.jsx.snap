// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderRoute render component using custom config name 1`] = `
<div>
  Hello World!!!
</div>
`;

exports[`renderRoute render component using default config name 1`] = `
<div>
  Hello World!!!
</div>
`;
